kahypar/partition/coarsening/heavy_edge_rater.h

```diff
@@ -128,6 +133,26 @@ class HeavyEdgeRater {
   HypernodeWeight thresholdNodeWeight() const {
     return _config.coarsening.max_allowed_node_weight;
   }
+  
+  void resetCommunities() {
+      _comm.assign(_comm.size(),0);
+  }
+  
+  void performLouvainCommunityDetection() {
+    HighResClockTimepoint start = std::chrono::high_resolution_clock::now();
+    std::set<ClusterID> distinct_comm;
+    EdgeWeight quality = _louvain.louvain();   
+    for(HypernodeID hn : _hg.nodes()) {
+        _comm[hn] = _louvain.clusterID(hn);
+        distinct_comm.insert(_comm[hn]);
+    }
+    HighResClockTimepoint end = std::chrono::high_resolution_clock::now();
+    std::chrono::duration<double> elapsed_seconds = end - start;
+    LOG("Louvain-Time: " << elapsed_seconds.count() << "s");
+    Stats::instance().addToTotal(_config,"louvainTime",elapsed_seconds.count());
+    Stats::instance().addToTotal(_config,"communities",distinct_comm.size());
+    Stats::instance().addToTotal(_config,"modularity",quality);
+  }
```

Comment:
Having this std::set here only for statistics seems to be overkill.
Please remove this. ;)